docs/how-to/providers/providers.md

```diff
@@ -0,0 +1,121 @@
+  |-------------------------|------------------------|----------------------------------------------------------------|
+  | `AWS_ACCESS_KEY_ID`     | `AKIA1234567890000000` | Access Key ID, with access to resources in Route 53            |
+  | `AWS_SECRET_ACCESS_KEY` | `Z01234567US0000000`   | Access Secret Access Key, with access to resources in Route 53 |
+  | `AWS_REGION`            | `eu-west-1`            | AWS Region                                                     |
+
+### GCP Cloud DNS 
+
+### Application Default Credentials (ADC)
+There are 2 types of methods that can be used to create credentials in the format that the MGC uses:
+1. User credentials provided by using the gcloud CLI
+2. Service account
+
+#### User credentials provided by using the cloud CLI
+
+1. Install Google cloud [cli](https://cloud.google.com/sdk/docs/install)
+2. Run the following [commands](https://cloud.google.com/docs/authentication/application-default-credentials#personal)
+
+#### Service Account
+
+To create a google service account please run the following [guide](https://cloud.google.com/docs/authentication/application-default-credentials#attached-sa)
+
+
+#### Setup a `./gcp-credentials.env` with credentials to access route 53
+
+  For example:
+
+``` bash
+GOOGLE={"client_id": "00000000-00000000000000.apps.googleusercontent.com","client_secret": "d-FL95Q00000000000000","refresh_token": "00000aaaaa00000000-AAAAAAAAAAAAKFGJFJDFKDK","type": "authorized_user"}
+PROJECT_ID: my_project_id
+
+```
+
+  | Env Var                 | Example Value          | Description                                                    |
+  |-------------------------|------------------------|----------------------------------------------------------------|
+  | `GOOGLE`     | `{"client_id": "00000000-00000000000000.apps.googleusercontent.com","client_secret": "d-FL95Q00000000000000","refresh_token": "00000aaaaa00000000-AAAAAAAAAAAAKFGJFJDFKDK","type": "authorized_user"}` |  This is the json created from either the credential created by the cli or the json from the Service account             |
+  | `PROJECT_ID` | `my_project_id`   | ID to the google project |
+
+#### Setup a `./controller-config.env` env-var file in the root of the repo with the following keys. Fill in your own values as appropriate. You will need access to a domain n in Cloud DNS in GCP:
+
+  ```bash
+  ZONE_ROOT_DOMAIN_GCP=jbloggs.google.hcpapps.net
+  LOG_LEVEL=1
+  ```
+
+  | Env Var                  | Example Value           | Description                                           |
+  |--------------------------|-------------------------|-------------------------------------------------------|
+  | `ZONE_ROOT_DOMAIN_GCP`       | `jbloggs.google.hcpapps.net`   | Hostname for the root Domain                          |
+  | `LOG_LEVEL`              | `1`                     | Log level for the Controller                          |
```

Comment:
i couldnt find a id for the zone all i can find is zone name and DNS name so i don't think they have one unless its hidden. So i have that the id is the same as domain name following the way we do when we create the mz from scratch. I can add in the that we need a id but it most cases i would think for dev they would be the same value

Reply:
The zone name in google dns is the unique id of the zone, and the value we should be setting in the MZ spec `spec.ID`. Does make me realise that we should probably not in that case be requiring or even using dnsName in the spec instead that should be retrieved from the status. 

Regardless, we can't assume they have created zones with names based on the dnsName. I could create a zone called `foobarbaz` with a dnsName of `not.foo.bar.baz.google.hcpapps.net` .

It might also be better to just name these env vars based on how that provider names things:
```
GCP_ZONE_NAME=foobarbaz 
GCP_ZONE_DNS_NAME=not.foo.bar.baz.google.hcpapps.net
```

Ends up as:
```
apiVersion: kuadrant.io/v1alpha1
kind: ManagedZone
metadata:
  name: not.foo.bar.baz.google.hcpapps.net
  namespace: multi-cluster-gateways
spec:
  id: foobarbaz
  domainName: not.foo.bar.baz.google.hcpapps.net
...
```

Reply:
Ok ill make those changes with the dnsname, and the changes with the env var